### PR TITLE
Improve errors for missing external instances in preview mode

### PR DESCRIPTION
**Pull request by eyelidlessness**
Addresses #363 but I'm not sure this is sufficient to close it, for a few reasons:

1. Currently this only addresses both preview cases. In the linked issue, @MartijnR says the intent is to:

    > show external file errors together with any form load errors, but still show the form (and disable submissions)

    Which seems to suggest this change should also address non-preview modes, but the change associated with that comment specifically special-cases preview mode, so I'm unclear about the intent. (This is why I'm also initially opening this as a draft.)

2. The error messages are _better_, but maybe not great? But those errors are now coming from enketo-core, so further improvement will likely need to happen there.

#### I have verified this PR works with

-   [x] Online form submission
-   [x] Offline form submission
-   [x] Saving offline drafts
-   [x] Submitting offline drafts
-   [x] Editing submissions
-   [x] Form preview
-   [ ] None of the above

#### What else has been done to verify that this works as intended?

Manual validation that there are at least somewhat meaningful errors in both preview modes, and that the form preview loads in both cases.

#### Why is this the best possible solution? Were any other approaches considered?

It may not be, but it's an expedient one. In discussion with @lognaturel we've talked about having a general classification of error conditions, which I'd definitely like to move towards as we learn more about them. I'd also like to consider adopting [`AggregateError`](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/AggregateError) for cases like this where there may be multiple errors, but that would impact `enketo-webform*.js` in ways I'm hesitant to touch without test coverage.

#### How does this change affect users? Describe intentional changes to behavior and behavior that could have accidentally been affected by code changes. In other words, what are the regression risks?

Following #396, I've made manual style changes to address warnings on the affected code. It's possible I made a mistake, which feels like the most significant risk in this change.

#### Do we need any specific form for testing your changes? If so, please attach one.

N/A

### CODE REVIEW

**Comment by lognaturel** [comment]
> this change should also address non-preview modes

Only handling previews feels ok to me. Users are strongly encouraged to go through a preview round.

> The error messages are better, but maybe not great? 

Could you please add a quick screenshot?

**Comment by eyelidlessness** [comment]
> Only handling previews feels ok to me. Users are strongly encouraged to go through a preview round.

Great, I'll set this ready for review then 🙂 

> Could you please add a quick screenshot?

image

All of these errors are coming from core. The first and last are essentially side-effects of the real failures to access external instances.

File: public/js/src/module/connection.js [hunk 1]
```diff
@@ -404,6 +396,70 @@ const getTransformURL = (basePath, enketoId) => {
...
+                const xml = await getDataFile(src, survey.languageMap);
+
+                return {
+                    ...instance,
+                    xml,
+                };
+            } catch (error) {
+                tasks.splice(index, 1);
+
+                if (options.isPreview) {
```
Comment (lognaturel) [inline public/js/src/module/connection.js:440]: This is the substantive change here, it unifies the two preview modes. Passed through from `getFormParts`

**Review by lognaturel — DISMISSED** [review]

**Review by lognaturel — APPROVED** [review]
I believe everything I was looking for was addressed by https://github.com/enketo/enketo-core/pull/873

**Comment by MartijnR** [comment]
> Currently this only addresses both preview cases. In the linked issue, @MartijnR says the intent is to:

> show external file errors together with any form load errors, but still show the form (and disable submissions)

> Which seems to suggest this change should also address non-preview modes, but the change associated with that comment specifically special-cases preview mode, so I'm unclear about the intent. (This is why I'm also initially opening this as a draft.)

Sorry I missed this. I believe that comment was referring to an OpenClinica customization where they would like the submit button (in non-preview views) to be disabled/removed in addition to showing the errors. We could continue discussing at https://github.com/enketo/enketo-express/issues/422 or its forthcoming PR perhaps.

**Pull request merged**